src/test/java/org/jenkinsci/plugins/envinject/EnvInjectPasswordTest.java

```diff
@@ -53,10 +62,103 @@ public void testEnvInjectJobParameterPassword() throws Exception {
         checkEnvInjectResult(build);
     }
 
+    @Bug(28409 )
+    public void testFileHandlesLeak() throws Exception {
+        final EnvInjectPasswordWrapper passwordWrapper = new EnvInjectPasswordWrapper();
+        passwordWrapper.setPasswordEntries(new EnvInjectPasswordEntry[]{
+                new EnvInjectPasswordEntry(PWD_KEY, PWD_VALUE)
+        });
+        final FileLeakBuildWrapper fileLeakDetector = new FileLeakBuildWrapper();
+
+        project.getBuildWrappersList().add(fileLeakDetector);
+        project.getBuildWrappersList().add(passwordWrapper);
+        
+        @SuppressWarnings("deprecation")  
+        FreeStyleBuild build = project.scheduleBuild2(0, new Cause.UserCause()).get();
+        Assert.assertEquals(Result.SUCCESS, build.getResult());
```

Comment:
Tip:

```java
buildAndAssertSuccess(project);
```

As intermediate steps, you can omit the `Cause`, use `assertBuildStatusSuccess`, etc.